### PR TITLE
Fix and Improve the SAM init wizard

### PR DESCRIPTION
**Pull request by hunterwerlla**
- The last PR broke the wizard because validation will always fail since `as? Runtime` no longer works (I missed this because Rider does no validation :( )
- add `minSamInit` and `minSamDebugging` to LambdaRuntime. They can't be `SemVer` since that comes from the intellij sdk and they are in core, so there are extension functions for that
  - With this, make `RuntimeGroup` based off of `LambdaRuntime` which allows us to remove hacks in the wizard for dotnet

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt [hunk 1]
```diff
@@ -147,7 +147,7 @@ class SamInitSelectionPanel(
      * Updates UI fragments in the wizard after a combobox update
      */
     private fun wizardUpdate() {
-        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        val selectedRuntime = runtimeComboBox.selectedItem as? LambdaRuntime
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt:150]: If you go through the model (`runtimes`), you dont need the cast

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt]: Changed

File: core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt [hunk 2]
```diff
@@ -5,19 +5,24 @@ package software.aws.toolkits.core.lambda
...
+    private val runtime: Runtime?,
+    val minSamInit: String? = null,
+    val minSamDebugging: String? = null,
+    private val runtimeOverride: String? = null
+) {
     NODEJS10_X(Runtime.NODEJS10_X),
     NODEJS12_X(Runtime.NODEJS12_X),
     JAVA8(Runtime.JAVA8),
-    JAVA8_AL2(Runtime.JAVA8_AL2),
+    JAVA8_AL2(Runtime.JAVA8_AL2, minSamDebugging = "1.2.0"),
```
Comment (abrooksv) [inline core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt:17]: Our min version is 1.13 right now, we can remove this

File: core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt [hunk 3]
```diff
@@ -5,19 +5,24 @@ package software.aws.toolkits.core.lambda
...
     JAVA11(Runtime.JAVA11),
     PYTHON2_7(Runtime.PYTHON2_7),
     PYTHON3_6(Runtime.PYTHON3_6),
     PYTHON3_7(Runtime.PYTHON3_7),
     PYTHON3_8(Runtime.PYTHON3_8),
     DOTNETCORE2_1(Runtime.DOTNETCORE2_1),
     DOTNETCORE3_1(Runtime.DOTNETCORE3_1),
-    DOTNET5_0(null, "dotnet5.0");
+    DOTNET5_0(null, minSamInit = "1.16.0", minSamDebugging = "1.13.0", runtimeOverride = "dotnet5.0");
```
Comment (rli) [inline core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt:25]: add quick comment? this makes sense but also funny how the versions seem backwards

**Review by rli — APPROVED** [review]

File: core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt (hunk 3 above)
Comment (hunterwerlla) [reply core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt]: It's actually no longer needed, removed

**Pull request merged**